core/primitives/src/utils.rs

```diff
@@ -112,6 +136,28 @@ impl KeyForAccessKey {
         key
     }
 
+    pub fn parse_account_id<K: AsRef<[u8]>>(raw_key: K) -> Result<AccountId, std::io::Error> {
+        let account_id_prefix =
+            KeyForColumnAccountId::parse_account_id_prefix(col::ACCESS_KEY, raw_key.as_ref())?;
+        let public_key_position = if let Some(index) =
+            account_id_prefix.iter().enumerate().find(|(_, c)| **c == 2).map(|(index, _)| index)
```

Comment:
XXX! This needs to be fixed!

@nearmax Currently, KeyForAccessKey is constructed as:

```
<col::ACCESS_KEY> + <account_id> + <borsh-serialized public_key>
```

Thus, currently, I assume that the borsh-serialized public key would start with non-printable char (my experiment showed that it starts with a byte `0x02`), but this is not really reliable. Any idea how we can do it better?

Reply:
It will not be an issue after https://github.com/nearprotocol/nearcore/pull/2244 because `hash(account_id)` length will be fixed 32 bytes. It also removes all separators